### DISCUSSION TITLE
I can't build apk

### DISCUSSION THREAD

**Discussion by johnied** [flet-dev/flet#4245] [Q&A] [answered]
### Question

Hello community,

This is the first time I am trying to make an .apk with flet build. I have installed ANDROID Studio and fixed all flutter issues (Flutter Doctor says No Issues Found!). I have added the following paths to Environment Variables:
C:\Users\[user]\AppData\Local\Android\Sdk\cmdline-tools\latest\bin
C:\Users\[user]\AppData\Local\Android\Sdk\platform-tools
C:\Users\[user]\AppData\Local\Android\Sdk\build-tools\35.0.0

The only packages I am importing to my app is flet and requests. So the requirement.txt contains the following:
flet>=0.22.0
requests==2.32.3

I am using flet build apk -vv and the process freezes at the following point:

VERBOSE: Using cached urllib3-2.2.3-py3-none-any.whl.metadata (6.5 kB)
(     ●) Packaging Python app ⏳... VERBOSE: Collecting certifi>=2017.4.17 (from requests==2.32.3)
VERBOSE: Using cached certifi-2024.8.30-py3-none-any.whl.metadata (2.2 kB)
(    ● ) Packaging Python app ⏳... VERBOSE: Collecting flet-runtime==0.24.1 (from flet-embed>=0.22.0)
VERBOSE: Using cached flet_runtime-0.24.1-py3-none-any.whl.metadata (1.1 kB)
(  ●   ) Packaging Python app ⏳... VERBOSE: Collecting flet-core==0.24.1 (from flet-runtime==0.24.1->flet-embed>=0.22.0)
VERBOSE: Using cached flet_core-0.24.1-py3-none-any.whl.metadata (971 bytes)
( ●    ) Packaging Python app ⏳... VERBOSE: Collecting httpx<1,>=0 (from flet-runtime==0.24.1->flet-embed>=0.22.0)
VERBOSE: Using cached httpx-0.27.2-py3-none-any.whl.metadata (7.1 kB)
(●     ) Packaging Python app ⏳... VERBOSE: Collecting oauthlib<4.0.0,>=3.2.2 (from flet-runtime==0.24.1->flet-embed>=0.22.0)
VERBOSE: Using cached oauthlib-3.2.2-py3-none-any.whl.metadata (7.5 kB)
( ●    ) Packaging Python app ⏳... VERBOSE: Collecting repath<0.10.0,>=0.9.0 (from flet-core==0.24.1->flet-runtime==0.24.1->flet-embed>=0.22.0)
VERBOSE: Using cached repath-0.9.0-py3-none-any.whl.metadata (899 bytes)
(  ●   ) Packaging Python app ⏳... VERBOSE: Collecting anyio (from httpx<1,>=0->flet-runtime==0.24.1->flet-embed>=0.22.0)
VERBOSE: Using cached anyio-4.6.2.post1-py3-none-any.whl.metadata (4.7 kB)
(    ● ) Packaging Python app ⏳... VERBOSE: Collecting httpcore==1.* (from httpx<1,>=0->flet-runtime==0.24.1->flet-embed>=0.22.0)
VERBOSE: Using cached httpcore-1.0.6-py3-none-any.whl.metadata (21 kB)
(     ●) Packaging Python app ⏳... VERBOSE: Collecting sniffio (from httpx<1,>=0->flet-runtime==0.24.1->flet-embed>=0.22.0)
VERBOSE: Using cached sniffio-1.3.1-py3-none-any.whl.metadata (3.9 kB)
VERBOSE: Collecting h11<0.15,>=0.13 (from httpcore==1.*->httpx<1,>=0->flet-runtime==0.24.1->flet-embed>=0.22.0)
VERBOSE: Using cached h11-0.14.0-py3-none-any.whl.metadata (8.2 kB)
(    ● ) Packaging Python app ⏳... VERBOSE: Collecting six>=1.9.0 (from repath<0.10.0,>=0.9.0->flet-core==0.24.1->flet-runtime==0.24.1->flet-embed>=0.22.0)
VERBOSE: Using cached six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
VERBOSE: Using cached requests-2.32.3-py3-none-any.whl (64 kB)
(   ●  ) Packaging Python app ⏳... VERBOSE: Using cached flet_embed-0.24.1-py3-none-any.whl (3.2 kB)
VERBOSE: Using cached flet_runtime-0.24.1-py3-none-any.whl (19 kB)
VERBOSE: Using cached flet_core-0.24.1-py3-none-any.whl (412 kB)
VERBOSE: Using cached certifi-2024.8.30-py3-none-any.whl (167 kB)
VERBOSE: Using cached charset_normalizer-3.4.0-py3-none-any.whl (49 kB)
VERBOSE: Using cached idna-3.10-py3-none-any.whl (70 kB)
VERBOSE: Using cached urllib3-2.2.3-py3-none-any.whl (126 kB)
VERBOSE: Using cached httpx-0.27.2-py3-none-any.whl (76 kB)
VERBOSE: Using cached httpcore-1.0.6-py3-none-any.whl (78 kB)
VERBOSE: Using cached oauthlib-3.2.2-py3-none-any.whl (151 kB)
VERBOSE: Using cached repath-0.9.0-py3-none-any.whl (4.7 kB)
VERBOSE: Using cached anyio-4.6.2.post1-py3-none-any.whl (90 kB)
VERBOSE: Using cached sniffio-1.3.1-py3-none-any.whl (10 kB)
VERBOSE: Using cached h11-0.14.0-py3-none-any.whl (58 kB)
VERBOSE: Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
(    ● ) Packaging Python app ⏳... VERBOSE: Installing collected packages: urllib3, sniffio, six, oauthlib, idna, h11, charset-normalizer, certifi, requests, repath, httpcore, anyio, httpx, flet-core, flet-runtime, flet-embed
(  ●   ) Packaging Python app ⏳...

Can sombody help me understand as to why it freezes after installing collected packages?
What is the next step of the build process? 
It seems that it cannot complete the next step, but no error is generated, it just goes forever.

Thank you in advance for your time.

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by LegendaryPistachio** [comment]
这个命令打包成功率，比较低。有个本地打包安卓应用的国家就好了，可以改变这一现象

**Reply by LegendaryPistachio** [reply]
The success rate of this command for packaging is quite low. It would be great to have a local solution for packaging Android apps, as that could change this situation.

**Reply by johnied** [reply]
Indeed, but in my case I think there is something wrong with my setup. It uses ony flet and requests.
It should be compiling normally.

Is there a step by step guide to follow, so that I perform all the steps correctly?

Thank you very much.

**Comment by johnied** [comment, marked as answer]
The problem was with flet 0.24.1. After installing flet 0.25.0.dev3570, the app compiled succesfully.
